File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/ui/base/BasePresenter.java
package com.example.ama.android2_lesson03.ui.base;

/**
 * Base implementation for {@link Presenter}
 *
 * @param <T> view for work with
 */
abstract public class BasePresenter<T extends PocketMapView> implements Presenter<T> {
    protected T view;

    @Override
    public void attachView(T view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/ui/base/BasePresenter.kt
package com.example.ama.android2_lesson03.ui.base

/**
 * Base implementation for [Presenter]
 *
 * @param <T> view for work with
 */
abstract class BasePresenter<T : PocketMapView> : Presenter<T> {

    protected var view: T? = null

    override fun attachView(view: T) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }
}